scripts/pipeline/orthology_benchmark.py

```diff
@@ -307,8 +307,48 @@ def run_orthology_tools(input_dir: str, orthofinder_params: str) -> None:
         raise RuntimeError(msg) from exc
 
 
-def prep_input_for_goc():
-    """Docstring"""
+def extract_orthologs(input_dir: str, species1: str, species2: str) -> List[Tuple[str, str]]:
+    """Returns a list of putative orthologous pairs inferred by OrthoFinder for a specified pair
+    of species.
+
+    If there are results for multiple OrthoFinder runs, the function uses those in the most recently modified
+    `Results_*` directory. The rationale behind this choice is that once OrthoFinder completes, the pipeline
+    automatically proceeds with calculating GOC scores for the predictions. At that point, it makes more sense
+    to take the most recent results than older ones. This also allows running the pipeline without user
+    intervention.
+
+    Args:
+        input_dir: Path to the directory with OrthoFinder output.
+        species1: Name of one species of interest.
+        species2: Name of another species of interest.
+
+    """
+    orthologs = []
+
+    orthofinder_res = [f.path for f in os.scandir(os.path.join(input_dir, "OrthoFinder"))
+                           if f.is_dir() and f.name.startswith("Results")]
+
+    latest_res = max(orthofinder_res, key=os.path.getmtime)
+
+    predictions_subdir = [f.path for f in os.scandir(os.path.join(latest_res, "Orthologues"))
+                          if f.is_dir() and re.match(f"(.*){species1}_core_(.*)", f.name)][0]
+    predictions_file = [f.path for f in os.scandir(predictions_subdir) if f.is_file() and
+                        re.match(f"(.*){species1}_core_(.*){species2}_core_(.*)", f.name)][0]
+
+    with open(predictions_file) as file_handler:
+        for line in file_handler:
+            if "Orthogroup" in line:
+                continue
+
+            temp = line.replace("\n", "").split("\t")
+            genes1 = temp[1].replace(" ", "").split(",")
+            genes2 = temp[2].replace(" ", "").split(",")
```

Comment:
This seems like a great opportunity to use the csv module (`import csv`). The DictReader class is particularly useful here, as it will consume the initial header line and use the headings as keys to the dictionary it returns for every data row. Because of this, you could use `species_key1` and `species_key2` to access the respective fields. I've also suggested using the comma-space string as subfield delimiter to simplify the process of getting individual gene IDs.
```suggestion
    with open(predictions_file) as file_handle:
        reader = csv.DictReader(file_handle, delimiter="\t")
        for row in reader:
            genes1 = row[species_key1].split(", ")
            genes2 = row[species_key2].split(", ")
```